### PR TITLE
add gomaxprocs preset for 2 existing use cases

### PR DESCRIPTION
**Pull request by haitch**
the gomaxprocs already have been used, convert them to preset label
- kubernetes-sigs/kueue/kueue-presubmits-main.yaml
- kubernetes/sig-k8s-infra/trusted/releng/releng-trusted.yaml

### CODE REVIEW

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml [hunk 1]
```diff
@@ -1,3 +1,12 @@
+presets:
+- labels:
+    preset-gomaxprocs: "true"
+  env:
+  - name: GOMAXPROCS
+    valueFrom:
+      resourceFieldRef:
+        resource: limits.cpu
```
Comment (BenTheElder) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml:8]: this is very elegant, later I would like to make this a default setting for all jobs, but probably not until after 1.31 releases

Comment (BenTheElder) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml:8]: though, this will break if anyone sets non-integer values.

**Comment by BenTheElder** [comment]
see also https://github.com/kubernetes/test-infra/issues/29472

I think it might be best if we invest in having prow do this by default? we ideally want some code to round the value and make sure it's an integer number of CPUs

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml (hunk 1 above)
Comment (haitch) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml:8]: no, it won't

I did a test:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: ubuntu
spec:
  restartPolicy: Never
  containers:
  - name: millicore
    image: ubuntu:latest
    command: [/bin/bash, -c, echo $GOMAXPROCS] # round up to 1
    env:
    - name: GOMAXPROCS
      valueFrom:
        resourceFieldRef:
          resource: limits.cpu
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: "250m"
  - name: empty
    image: ubuntu:latest
    command: [/bin/bash, -c, echo $GOMAXPROCS] # phisical core number on the host, I got 16
    env:
    - name: GOMAXPROCS
      valueFrom:
        resourceFieldRef:
          resource: limits.cpu
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
  - name: millicore2048
    image: ubuntu:latest
    command: [/bin/bash, -c, echo $GOMAXPROCS] # round up to 3
    env:
    - name: GOMAXPROCS
      valueFrom:
        resourceFieldRef:
          resource: limits.cpu
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: "2048m"
  - name: point5
    image: ubuntu:latest
    command: [/bin/bash, -c, echo $GOMAXPROCS] # round up to 1
    env:
    - name: GOMAXPROCS
      valueFrom:
        resourceFieldRef:
          resource: limits.cpu
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: 0.5
  - name: 3point5
    image: ubuntu:latest
    command: [/bin/bash, -c, echo $GOMAXPROCS] # round up to 4
    env:
    - name: GOMAXPROCS
      valueFrom:
        resourceFieldRef:
          resource: limits.cpu
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: 3.5
  - name: two
    image: ubuntu:latest
    command: [/bin/bash, -c, echo $GOMAXPROCS] # print 2
    env:
    - name: GOMAXPROCS
      valueFrom:
        resourceFieldRef:
          resource: limits.cpu
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: 2
```

**Comment by haitch** [comment]
> see also #29472
> 
> I think it might be best if we invest in having prow do this by default? we ideally want some code to round the value and make sure it's an integer number of CPUs

1. define in presets without label need, just like [the goproxy one](https://github.com/kubernetes/test-infra/blob/00dc889b3de1cc9d39625433a8186d0ca266c1ea/config/prow/config.yaml#L1032-L1035), I am worry if there is some job doesn't have cpulimit defined, things can go wild.
2. define in global preset with a label, **I am going for this one**, we can provide value for any job with cpulimit set.
3. have prow do this by default, I think that's a boundary concern, I don't see downside, but no optout.

**Comment by haitch** [comment]
@ahg-g @puerco can you put ok-to-test label?

**Pull request closed without merging**